### ISSUE TITLE
Restrict Access Permissions option for Workspaces

### ISSUE THREAD

**Issue by peey** [benweet/stackedit#1313]
Currently, when you try to synchronize a single file using the "synchronize" option, there is a checkbox allowing you to "Restrict Access" in the popup that appears before you're redirected to the permissions authorization page

![image](https://user-images.githubusercontent.com/4688870/39923284-a87687fc-5540-11e8-8047-90e86a813b64.png)

Is it possible to also restrict access when adding a new Google Drive Workspace?

Basically what I want is that stackedit should be granted permission to do whatever it likes within a folder of my choosing within my drive and have permissions to no other files

My use case is basically https://github.com/benweet/stackedit/issues/748

**Comment by benweet** [comment]
That's not possible I'm afraid. StackEdit wouldn't be allowed to open shared workspaces.

**Comment by rainabba** [comment]
Effectively you're taking the whole "request access via email" flow and making it instant with that prompt and the access I'd expect it to grant you. I am asking sincerely btw, not trolling, I promise. I just hit the same issue I think as I was trying to open an .md file from the Drive interface. I couldn't when I checked that open because the page had a bad promise call to .then(). I'm guessing it's the list() call results and they don't exist when it's checked.

**Comment by rainabba** [comment]
...then again, I might be seeing something totally unrelated because apparently it doesn't work even when I do check it. Is StackEdit compatible with Team Drives?
